### ISSUE TITLE
"Finish" wasmcloud-chat example

### ISSUE THREAD

**Issue by autodidaddict** [wasmCloud/examples#41]
The wasmcloud chat example is currently incomplete, largely due to the fact that we had to record our demo videos for upcoming conferences well in advance of reaching feature completion. The following is a list of what we think should be done before we can consider this sample "complete":

* [ ] - Telnet channel needs to prompt users for their username and password to log in or prompt "new" for new user
* [ ] - NATS channel needs to offer a topic for presenting username and password in exchange for a bearer token, as well as a topic for creating a new user
* [ ] - Internal plumbing needs to require the use of access/bearer tokens to illustrate real-world security in a real app
* [ ] - Presence actor needs to be utilized for reporting and querying "who's  online" (e.g. support `/who` from telnet)
* [ ] - Telnet and NATS channel need to provide mechanisms for joining and removing rooms
* [ ] - Telnet and NATS channel need to provide mechanisms for querying chat history for a given room
* [ ] - NATS channel needs a user-friendly front-end (web app? another CLI? mobile app?)

**Comment by autodidaddict** [comment]
Checking with @brooksmtownsend to see if I missed anything in the todo list.

**Comment by brooksmtownsend** [comment]
I like the web front-end option for the NATS channel, which essentially allows us to have 3 channels if you consider manual NATS publishing an acceptable channel.

Only additional consideration atm:
- [ ] What happens when a user is logged into a single channel and they log into another channel. Are we enforcing single sessions or allowing all sessions to receive a message?

**Comment by autodidaddict** [comment]
If you look at the presence data we're using in the `presence` actor, you should find that we're aggregating user+channel, which means when you get a list of who's "present", you get the user and then _all of the channels_ from which they've received a presence track event that hasn't yet expired.

tl;dr yes, I can be "online" from both the telnet and the nats channels at the same time.

**Comment by brooksmtownsend** [comment]
Closing this, as `wasmcloudchat` has joined the others in the demo graveyard 💀
